plenum/common/event_bus.py

```diff
@@ -0,0 +1,15 @@
+from collections import defaultdict
+from typing import Callable, Type, Dict, List
+
+
+class EventBus:
+    def __init__(self):
+        self._handlers = defaultdict(list)  # type: Dict[Type, List[Callable]]
+
+    def subscribe(self, message_type: Type, handler: Callable):
```

Comment:
Should we define the interface for subscribers (observers)?

Reply:
I don't think so. In this generic case interface is as simple as callable accepting arbitrary parameters. The only limitation is that these parameters should match those passed to `send`